src/frontend/src/lib/components/send/QRCodeScan.svelte

```diff
@@ -0,0 +1,96 @@
+<script lang="ts">
+	import { QRCodeReader } from '@dfinity/gix-components';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { decodeQrCode } from '$lib/utils/qr-code.utils';
+	import { nonNullish } from '@dfinity/utils';
+	import type { Token } from '@dfinity/utils';
+	import type { QrStatus } from '$lib/types/qr-code';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+
+	export let expectedToken: Token;
+	export let destination: string | undefined;
+	export let amount: number | undefined;
+
+	const dispatch = createEventDispatcher();
+
+	let timeoutId: NodeJS.Timeout | undefined;
```

Comment:
Given that this PR aims to be a refactoring only, can you provide the feature `timeout`, which seems to be a newest thing,  and its motivation in a separate PR please?

Reply:
done